### ISSUE TITLE
[QUESTION] Did u compare ssl pretrained with coco pretrained for object detection?

### ISSUE THREAD

**Issue by Serhii-Tiurin** [lightly-ai/lightly-train#71]
First, I’d like to say thank you — you’re doing a great job, and I truly appreciate the work and effort behind this project. 
Question is, did u make comparison with ssl pretrained models and coco pretrained models for object detection?

I have dataset of 30k images of person in hardhat, person without hardhat.
Environment is noisy manufacturing, so it is a big difference from usual coco images in terms of distribution.
First, on unlabeled images trained according to this tutorial trained ssl checkpoint https://docs.lightly.ai/train/stable/models/ultralytics.html

Then trained 2 models, both on 10% of labeled data, one with exported_last.pt from ssl, one with coco pretrained, all other hyperparameters are the same, same full 100% validation.
And i still got better metric with coco, like with 8% F1 difference.
Decreasing amout of labeled data to 2-3-5% didnt help.

**Comment by stegmuel** [comment]
Thanks for your interest in our work.

Great question, we are actively working on this. So far, we've observed that whether or not we can outperform off-the-shelf COCO weights with SSL pre-training depends heavily on the experimental setup, e.g. the degree of domain shift, the number of images, etc.

Importantly, in the tutorial you referred to, the backbone is randomly initialized, and so is the detection head. Since the head accounts for a significant portion of the model's parameters, having it randomly initialized is a considerable disadvantage. You might want to try pre-training starting from the COCO weights instead:

```
import lightly_train

if __name__ == "__main__":
    lightly_train.train(
        out="out/my_experiment",  
        data="my_data_dir",
        model="ultralytics/yolov8s.pt",  # <----- Note the change here
    )
```

We're also investigating this setting more deeply on our side.

Looking forward to hearing back from you!

**Comment by XevenQC** [comment]
Any progress for this issue? I've encountered the same issue.

If need pretrain from coco weights, what's the advantage of ssl compared to transfer learning?

**Comment by stegmuel** [comment]
Thanks for reaching out!

Did you also try pretraining on top of the COCO weights? We've seen promising results in that setup too. it can work well depending on the domain.

Ideally, self-supervised learning and off-the-shelf initialization (like COCO weights) should be viewed as complementary, not mutually exclusive.

We had to reduce work on this due to higher-priority tasks, but we've actually just started picking it back up again.

**Comment by bluesy7585** [comment]
Hi lightly-train team,

Thanks for your great job on this project! It's really nice to have a self supervised learning tool that can easily integrate with YOLO training pipeline.

I have tried few quick experiments on ssl from COCO pretrained weight using the code from tutorial.
our unlabel data is a small dataset for human detection (<25K)
```
import lightly_train

if __name__ == "__main__":
    lightly_train.train(
        out="lightly_train_out/exp_11s", 
        data="/home/jovyan/new_repo/ultralytics/humanDet_ssl_ln",
        model="ultralytics/yolo11s.pt",
        method="distillation"     
    )
```
after ssl, finetune the exported weights using ultralytics command line on 2k images. (and 500 validation images)
`yolo train data=human_det.yaml model=/home/jovyan/new_repo/lightly_train_out/exp_11s/exported_models/exported_last.pt epochs=60
`
the performance I got from the finetuned model is basically the same with simply train from COCO pretrain.
ie.
`yolo train data=human_det.yaml model=yolo11s.pt epochs=60
`

I wonder that can we get benefits from ssl if we train the ssl longer under this setup? or should we train ssl from scratch?

**Comment by yutong-xiang-97** [comment]
Hi @bluesy7585, thank you for your question!

May we have your training logs with your training configs? This way, we can help you debug and find what we could do to improve the pretraining performance.

**Comment by bluesy7585** [comment]
Hi @yutong-xiang-97 [here ](https://drive.google.com/file/d/1tsbe0uP4B2Ep2r_PtdiDzjjALnOBKlog/view?usp=drive_link)is the log file in output folder.

**Comment by yutong-xiang-97** [comment]
Hi @bluesy7585, thank you for providing the training log!

We noticed that you only trained with 100 epochs, which may not be sufficient for a performance boost considering a ~25k dataset is quite small. We suggest increasing the training epochs as much as you can. Check our [docs](https://docs.lightly.ai/train/stable/methods/distillation.html#lightly-recommendations) for suggestions for training with distillation.

Hope this helps!

**Comment by bluesy7585** [comment]
I just finished a 1000 epochs ssl training (from scratch) and got the same issue as original poster.
the model finetuned from ssl exported is no better than train from COCO pretrained.

the data used for ssl is the same domain with finetune dataset (human detection)
since our dataset (~=25k) is much smaller than COCO. so I think its not a fair comparison.

Can anyone provide a reference information about how many images (and epochs) is sufficient to outperformed COCO pretrained?

**Comment by Bovey0809** [comment]
I have the same issue, the model finetuned from dinov3 is not as good as from COCO pretrained.

**Comment by Jordan-Pierce** [comment]
I'm doing something similar and tested how distillation may or may not improve model performance on down-stream tasks (object detection). I found the following:

- yolo11n trained from scratch (mAP 0.533)
- yolo11n distilled from scratch, then fine-tuned (mAP 0.445)
- yolo11n (starting from coco weights) fine-tuned (i.e., baseline) (mAP 0.578)
- yolo11n (starting from coco weights) distilled, then fine-tuned (mAP 0.607)

So in my case there was actually a performance gain by starting with pre-trained coco weights, distilling, then fine-tuning on top of those. But when increasing the data by an order of magnitude, I did not see gains (actually losses, mAP 0.579). 

I came here to actually see what the recommendations are on training parameters (specifically for ultralytics) when fine-tuning using distilled weights. The official recommendations on fine-tuning (as seen [here](https://www.reddit.com/r/Ultralytics/comments/1eolwl8/the_correct_way_to_train_from_a_previously/#:~:text=When%20continuing%20from%20a%20fine,4%20Go%20to%20comments%20Share)) actually specify how to adjust the parameters when using a custom model as a starting point, as opposed to using the default coco weights. I'm currently following the recommendations (i.e., treating these distilled weights as custom weights), but I'm wondering if I should set the training parameters as if these distilled weights are "from scratch". 

```python
# Instructions if fine-tuning on top of existing model weights, not COCO
params['warmup_epochs'] = 0
params['lr0'] = 0.0005  # something small, matching the end lr of previous session
params['optimizer'] = 'AdamW'  # must manually set instead of relying on 'auto'
```
